### PR TITLE
Add `strict` option

### PR DESCRIPTION
**Pull request by franciscocardoso**
This commit also contains code style updates, documentation improvements
and a rename to `ssn-validator`.

Refs: https://github.com/seegno/is-valid-ssn/issues/1

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,3 +2,4 @@ language: node_js
 
 node_js:
   - 4
+  - 5
```
Comment (ruimarinho) [inline .travis.yml]: Please convert to docker-compose.

File: README.md [hunk 2]
```diff
@@ -2,43 +2,84 @@
 
 [![build status][travis-image]][travis-url]
```
Comment (ruimarinho) [inline README.md]: I would move this to another subtopic (`Status`).

File: README.md [hunk 3]
```diff
@@ -2,43 +2,84 @@
...
 
-This modules allows you to check if a number is a valid.
+This modules provides some tools to handle `Social Security Numbers` from the `United States of America`.
 
 ## Installation
```
Comment (ruimarinho) [inline README.md:7]: I would keep this simple:

```
## Installation
Install the package via `npm`:

npm install ssn-validator --save
```

File: README.md [hunk 4]
```diff
@@ -2,43 +2,84 @@
...
 
-*NOTE:* The input number **must not** be formatted to `xxx-xxxxxx`.
+### `isValid(value, [options])`
 
-> Check if number is valid.
+This method validates if the given value is a valid `Social Security Number`.
 
+#### Arguments
+
+1. `value` *(&#42;)*: The value to check.
```
Comment (ruimarinho) [inline README.md]: to validate.

File: README.md [hunk 5]
```diff
@@ -2,43 +2,84 @@
...
+2. `[options]` *(Object)*: The options object.
+3. `[options.strict=true]` _(boolean)_: Whether or not to reject formatted numbers.
```
Comment (ruimarinho) [inline README.md]: `Whether or not formatting characters such as dashes or spaces should be accepted.`

File: README.md [hunk 6]
```diff
@@ -2,43 +2,84 @@
...
+
+#### Returns
+*(boolean)*:  Returns `true` if `value` is a valid `Social Security Number`, else `false`.
```
Comment (ruimarinho) [inline README.md]: Perhaps remove backticks around Social Security Number?

Comment (ruimarinho) [inline README.md]: Perhaps remove backticks around Social Security Number?

File: README.md [hunk 7]
```diff
@@ -2,43 +2,84 @@
 
 [![build status][travis-image]][travis-url]
 
-This modules allows you to check if a number is a valid.
+This modules provides some tools to handle `Social Security Numbers` from the `United States of America`.
```
Comment (ruimarinho) [inline README.md]: Same here - remove backticks for Social Security Numbers?

File: README.md [hunk 8]
```diff
@@ -2,43 +2,84 @@
...
+mask('123-123-123', { strict: false });
+// => XXX-X23-123
+
+mask('123123123');
+// => XXXX23123
 ```
 
+* * *
+
 ## Running tests
```
Comment (ruimarinho) [inline README.md]: `Tests` only?

File: README.md [hunk 9]
```diff
@@ -2,43 +2,84 @@
...
+isValid('123-123-123');
+// => false
+
+isValid('123-123-123', { strict: false });
+// => true
+
+isValid('123123123');
+// => true
 ```
+* * *
```
Comment (ruimarinho) [inline README.md]: `--------------------------------------------------------------------------------`

File: README.md [hunk 10]
```diff
@@ -2,43 +2,84 @@
...
+This method will help you protect this sensitive piece of information by obfuscating some digits.
 
+#### Arguments
+
+1. `value` *(&#42;)*: The value to mask.
+2. `[options]` *(Object)*: The options object.
+3. `[options.strict=true]` _(boolean)_: Whether or not to reject formatted values.
+
+#### Returns
+*(string)*:  Returns a masked value.
```
Comment (ruimarinho) [inline README.md]: Extra space.

Comment (ruimarinho) [inline README.md]: `Returns the masked value`?

File: README.md [hunk 11]
```diff
@@ -2,43 +2,84 @@
...
+### `mask(value, [options])`
 
-> Mask the number.
+This method will help you protect this sensitive piece of information by obfuscating some digits.
 
+#### Arguments
+
+1. `value` *(&#42;)*: The value to mask.
+2. `[options]` *(Object)*: The options object.
+3. `[options.strict=true]` _(boolean)_: Whether or not to reject formatted values.
```
Comment (ruimarinho) [inline README.md]: Same description update as mentioned above?

File: README.md [hunk 12]
```diff
@@ -2,43 +2,84 @@
```
Comment (ruimarinho) [inline README.md:3]: Title should be:

```
# ssn-validator
```

File: package.json [hunk 13]
```diff
@@ -1,42 +1,53 @@
...
+  },
+  "main": "dist/index.js",
   "keywords": [
     "IRS",
     "SSN",
     "TIN"
   ],
   "options": {
-    "mocha": "--compilers js:babel-register --recursive --require should"
+    "mocha": "--compilers js:babel-register --recursive --require should test"
```
Comment (ruimarinho) [inline package.json]: You can remove `test` from here.

File: package.json [hunk 14]
```diff
@@ -1,42 +1,53 @@
...
+    "changelog": "github_changelog_generator --no-issues --header-label='# Changelog'",
+    "cover": "babel-node node_modules/.bin/isparta cover --report html _mocha -- $npm_package_options_mocha",
     "lint": "eslint src test && jscs src test",
-    "prepublish": "npm run lint && npm test && npm run build",
-    "test": "NODE_ENV=test mocha $npm_package_options_mocha"
-  },
-  "dependencies": {
-    "lodash": "3.10.1"
+    "prepublish": "npm run transpile",
+    "testdocker": "docker-compose run --rm sut",
```
Comment (ruimarinho) [inline package.json:32]: Missing the docker compose file.

File: package.json [hunk 15]
```diff
@@ -1,42 +1,53 @@
...
+    "test": "NODE_ENV=test mocha $npm_package_options_mocha",
```
Comment (ruimarinho) [inline package.json]: I think the NODE_ENV can be removed from here since we're not making use of it.

File: package.json [hunk 16]
```diff
@@ -1,42 +1,53 @@
...
-    "jscs-config-seegno": "1.1.0",
-    "mocha": "2.4.5",
-    "should": "8.2.1"
+    "babel-cli": "^6.4.0",
+    "babel-eslint": "^4.1.6",
+    "babel-plugin-add-module-exports": "^0.1.2",
+    "babel-preset-es2015-node4": "^2.0.2",
+    "babel-register": "^6.3.13",
+    "eslint": "^1.10.3",
+    "eslint-config-seegno": "^1.2.0",
```
Comment (ruimarinho) [inline package.json]: 2.0.0?

File: src/index.js [hunk 17]
```diff
@@ -1,58 +1,55 @@
...
 
-const blacklist = [
-  '078051120',
-  '219099999',
-  '457555462'
-];
-const expression = /^(?!666|000|9\d{2})\d{3}(?!00)\d{2}(?!0{4})\d{4}$/;
+const expressions = {
+  soft: /^(?!666|000|9\d{2})\d{3}-(?!00)\d{2}-(?!0{4})\d{4}$/,
+  strict: /^(?!666|000|9\d{2})\d{3}(?!00)\d{2}(?!0{4})\d{4}$/
```
Comment (ruimarinho) [inline src/index.js:22]: I don't think we need to add another expression for the soft case. Simply replacing the characters we would like to support (e.g. multiple spaces and dashes?) and then running the resulting value through the normal expression would suffice in my opinion.

File: test/index_test.js [hunk 18]
```diff
@@ -3,7 +3,7 @@
  * Module dependencies.
  */
 
-import { default as isValidSsn, mask } from '../src';
+import { isValid, mask } from '../src';
```
Comment (ruimarinho) [inline test/index_test.js]: Invert order (defaults first, then named).

File: test/index_test.js [hunk 19]
```diff
@@ -26,26 +26,30 @@ const validNumbers = [
  * Test.
```
Comment (ruimarinho) [inline test/index_test.js]:
```
Test `ssn-validator`.
```

File: test/index_test.js [hunk 20]
```diff
@@ -26,26 +26,30 @@ const validNumbers = [
...
+describe('SsnValidator', () => {
+  describe('isValid()', () => {
     it('should return `false` if ssn is invalid', () => {
       for (const ssn of invalidNumbers) {
-        isValidSsn(ssn).should.be.false();
+        isValid(ssn).should.be.false();
       }
     });
 
+    it('should return `false` if ssn is valid, formated, and `options.strict` is `true`', () => {
```
Comment (ruimarinho) [inline test/index_test.js]: if "value" instead of "if ssn"?

File: test/index_test.js [hunk 21]
```diff
@@ -26,26 +26,30 @@ const validNumbers = [
  * Test.
  */
 
-describe('isValidSsn', () => {
-  describe('default()', () => {
+describe('SsnValidator', () => {
+  describe('isValid()', () => {
     it('should return `false` if ssn is invalid', () => {
       for (const ssn of invalidNumbers) {
```
Comment (ruimarinho) [inline test/index_test.js]: const value?

Comment (ruimarinho) [inline test/index_test.js]: `const numbers = { valid: [], invalid: [] }`?

File: src/index.js (hunk 17 above)
Comment (franciscocardoso) [inline src/index.js:22]: @ruimarinho Don't you think that, this way, we are making a better assert?
If I understand correctly you suggestion will accept a number formatted like `111223333--` as valid and that is not the ssn mask/format.

**Comment by ruimarinho** [comment]
@franciscocardoso this should have been broken into multiple commits.

**Comment by ruimarinho** [comment]
Missing `.babelrc`?

File: .dockerignore [hunk 22]
```diff
@@ -0,0 +1,5 @@
+Changelog.md
```
Comment (ruimarinho) [inline .dockerignore]: CHANGELOG.md

File: .travis.yml [hunk 23]
```diff
@@ -1,4 +1,7 @@
-language: node_js
```
Comment (ruimarinho) [inline .travis.yml:1]: language: bash

File: .travis.yml [hunk 24]
```diff
@@ -1,4 +1,7 @@
...
+script:
+  - docker-compose run --rm sut
 
-node_js:
-  - 4
+services:
+  - docker
+
+sudo: false
```
Comment (ruimarinho) [inline .travis.yml]: Probably not needed.

File: Dockerfile [hunk 25]
```diff
@@ -0,0 +1,25 @@
+FROM mhart/alpine-node:5
+
+RUN apk add --update make gcc g++ python git && rm -rf /var/cache/apk/*
```
Comment (ruimarinho) [inline Dockerfile]: No module need to be compiled AFAIK.

File: README.md [hunk 26]
```diff
@@ -1,44 +1,85 @@
-# Social Security Number (SSN)
+# ssn-validator
 
-[![build status][travis-image]][travis-url]
+This modules provides some tools to handle Social Security Numbers from the `United States of America`.
```
Comment (ruimarinho) [inline README.md]: `Validate and mask U.S. Social Security Numbers.`

File: README.md [hunk 27]
```diff
@@ -1,44 +1,85 @@
...
+
+## Status
 
-This modules allows you to check if a number is a valid.
+[![build status][travis-image]][travis-url]
```
Comment (ruimarinho) [inline README.md]: Missing npm status.

File: README.md [hunk 28]
```diff
@@ -1,44 +1,85 @@
...
 
 ## Installation
+Install the package via `npm`:
+
+* npm: `npm install --save ssn-validator`
```
Comment (ruimarinho) [inline README.md]: Remove `* npm`?

File: README.md [hunk 29]
```diff
@@ -1,44 +1,85 @@
...
 ```sh
 npm test
 ```
 
-[travis-image]: https://img.shields.io/travis/seegno/is-valid-ssn.svg?style=flat-square
-[travis-url]: https://travis-ci.org/seegno/is-valid-ssn
+[travis-image]: https://img.shields.io/travis/seegno/ssn-validator.svg?style=flat-square
+[travis-url]: https://travis-ci.org/seegno/ssn-validator
 
 ## Credits
```
Comment (ruimarinho) [inline README.md:84]: Missing License.

File: src/index.js (hunk 17 above)
Comment (ruimarinho) [inline src/index.js:22]: No, my suggestion is simply adding optional "-" (dashes) and " " (spaces) to the original regex.

`111-222 333` would be valid, `111-222333-` would not.

Comment (ruimarinho) [inline src/index.js:22]: This may not be possible without using named groups.. might not be worth it.

File: test/index_test.js [hunk 30]
```diff
@@ -3,52 +3,69 @@
...
+    it('should return `false` if value is invalid', () => {
+      for (const value of numbers.invalid) {
+        isValid(value).should.be.false();
       }
     });
 
-    it('should return `true` if ssn is valid', () => {
-      for (const ssn of validNumbers) {
-        isValidSsn(ssn).should.be.true();
+    it('should return `false` if value is valid, formated, and `options.strict` is `true`', () => {
```
Comment (ruimarinho) [inline test/index_test.js]: formatted

File: test/index_test.js [hunk 31]
```diff
@@ -3,52 +3,69 @@
...
+      isValid('011-23-4567').should.be.false();
+    });
+
+    it('should return `true` if value is valid, formated, and `options.strict` is `false`', () => {
```
Comment (ruimarinho) [inline test/index_test.js]: formatted

**Pull request merged**